npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -101,12 +101,21 @@ def npm_files
         fetched_npm_files
       end
 
+      def run_yarn_install
+        Dir.chdir("repo") do
+          # --refresh-lockfile    Refresh the package metadata stored in the lockfile
```

Comment:
Any reason why we are not using the Helper.Subprocess method?

Also, can we add tests?

Reply:
Based on the analysis, `yarn immutable install` is creating `.yarn/cache` directory on the repo directory. So I have updated fix and test accordingly.

With this, I ran the cli and ensured, Customer reported issue is fixed